src/limiters/limiters_dg3d.jl

```diff
@@ -0,0 +1,40 @@
+
+function limiter_zhang_shu!(u::AbstractArray{<:Any,5},
+                            threshold::Real, variable,
+                            mesh, equations, dg::DGSEM, cache)
+  @unpack weights = dg.basis
+
+  Threads.@threads for element in eachelement(dg, cache)
+    # dermine minimum value
+    value_min = typemax(eltype(u))
+    for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, j, k, element)
+      value_min = min(value_min, variable(u_node, equations))
+    end
+
+    # detect if limiting is necessary
+    value_min < threshold || continue
+
+    # compute mean value
+    u_mean = zero(get_node_vars(u, equations, dg, 1, 1, 1, element))
+    for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, j, k, element)
+      u_mean += u_node * weights[i] * weights[j] * weights[k]
```

Comment:
are we still sure, that it is not worth to make weights3d[i,j,k] instead of computing them (same for 2d)?

Reply:
Feel free to open an issue or make a PR and benchmark that difference there. I haven't done extensive benchmarks of this yet.

Reply:
I've opened an issue for that (https://github.com/trixi-framework/Trixi.jl/issues/279).